lib/phoenix/socket/transport.ex

```diff
@@ -388,6 +388,57 @@ defmodule Phoenix.Socket.Transport do
     end
   end
 
+  @doc """
+  Extracts connection information from the conn and return a map.
+
+  Keys are retrieved from the optional transport option `:connect_info`.
+  This functionality is transport specific. Please refer to your transports's
+  documentation for more information.
+
+  The supported keys are:
+
+    * `:peer_data` - the result of `Plug.Conn.get_peer_data/1`.
+    * `:x_headers` - a list of all request headers that have an "x-" prefix.
+    * `:uri` - a `%URI{}` derived from the conn.
+
+  """
+  def connect_info(conn, opts) do
+    keys = Keyword.get(opts, :connect_info, [])
+    
+    for key <- keys, into: %{} do
+      case key do
+        :peer_data ->
+          {:peer_data, Plug.Conn.get_peer_data(conn)}
+
+        :x_headers ->
+          {:x_headers, fetch_x_headers(conn)}
+
+        :uri ->
+          {:uri, fetch_uri(conn)}
+
+        _ ->
+          raise ArgumentError, "connection info keys are expected to be one of [:peer_data, :x_headers, :uri], got: #{inspect(key)}"
```

Comment:
Maybe it’s better to use here “:connect_info keys”, due to the key and function name?

And maybe “on of :peer_data, :x_headers, or :uri”?